fortnite_api/shop.py

```diff
@@ -98,31 +120,59 @@ class BrShopEntry:
     """
 
-    def __init__(self, data):
-        self.regular_price = data.get('regularPrice')
-        self.final_price = data.get('finalPrice')
-        self.discount = self.regular_price - self.final_price if self.regular_price and self.final_price else None
-        self.bundle = BrShopBundle(data.get('bundle')) if data.get('bundle') else None
-        self.banner = BrShopBanner(data.get('banner')) if data.get('banner') else None
-        self.giftable = data.get('giftable')
-        self.refundable = data.get('refundable')
-        self.sort_priority = data.get('sortPriority')
-        self.categories = data.get('categories')
-        self.section_id = data.get('sectionId')
-        self.section = BrShopSectionNew(data.get('section')) if data.get('section') else None
-        self.dev_name = data.get('devName')
-        self.offer_id = data.get('offerId')
-        self.display_asset_path = data.get('displayAssetPath')
-        self.tile_size = data.get('tileSize')
-        self.new_display_asset_path = data.get('newDisplayAssetPath')
-        self.new_display_asset = data.get('newDisplayAsset')
-        self.items = [BrCosmetic(item_data) for item_data in data.get('items')]
-        self.raw_data = data
+    __slots__: Tuple[str, ...] = (
+        'regular_price',
+        'final_price',
+        'bundle',
+        'banner',
+        'giftable',
+        'refundable',
+        'sort_priority',
+        'categories',
+        'section_id',
+        'section',
+        'dev_name',
+        'offer_id',
+        'display_asset_path',
+        'tile_size',
+        'new_display_asset_path',
+        'new_display_asset',
+        'items',
+        'raw_data',
+    )
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.regular_price: Optional[int] = data.get('regularPrice')
+        self.final_price: Optional[int] = data.get('finalPrice')
+
+        self.bundle: Optional[BrShopBundle] = (bundle := data.get('bundle')) and BrShopBundle(bundle)
+        self.banner: Optional[BrShopBanner] = (banner := data.get('banner')) and BrShopBanner(banner)
+        self.giftable: Optional[bool] = data.get('giftable')
+        self.refundable: Optional[bool] = data.get('refundable')
+        self.sort_priority: Optional[int] = data.get('sortPriority')
+        self.categories: Optional[List[str]] = data.get('categories')
+        self.section_id: Optional[int] = data.get('sectionId')
+
+        self.section: Optional[BrShopSection] = (section := data.get('section')) and BrShopSection(section)
+        self.dev_name: Optional[str] = data.get('devName')
+        self.offer_id: Optional[int] = data.get('offerId')
+        self.display_asset_path: Optional[str] = data.get('displayAssetPath')
+        self.tile_size: Optional[str] = data.get('tileSize')
+        self.new_display_asset_path: Optional[str] = data.get('newDisplayAssetPath')
+        self.new_display_asset: Optional[str] = data.get('newDisplayAsset')
```

Comment:
I thought this same thing. It's not pushed yet but 4 days ago I already wrote an `Asset` class. It will hold some logic for any Asset with a url - and optional path.

There is some work to be done due to the fact that some objects need `HTTPClient` passed and I haven't gotten around to it yet :)

![image](https://user-images.githubusercontent.com/75498301/171475370-23b2780d-f014-440b-960f-52533b816e05.png)
![image](https://user-images.githubusercontent.com/75498301/171475319-48aafe87-ea6b-4bec-ba8d-8f74a6a9a1d9.png)